R/Spectra-functions.R

```diff
@@ -1008,3 +1046,138 @@ filterPrecursorPeaks <- function(object, tolerance = 0, ppm = 20,
         object@processing, "Filter: remove peaks ", msg)
     object
 }
+
+#' @title Define a factor for parallel access to peaks data
+#'
+#' @description
+#'
+#' Define a factor to split a `Spectra` for parallel processing based on
+#' parallel processing setup,
+#'
+#' If `processingChunkSize(x)` is defined and its value is smaller then
+#' `length(x)`, a factor depending on this chunk size is returned.
```

Comment:
Why `length(x)` ? and not `max(lenghths(x))` ? So that if the user wants to use a chunksize smaller than the number spectra per file they can but otherwise do it files by files.. idk if that make sense... 
And i realise that only really make sense if the backend is `MsBackendMzR`...

Reply:
I know my comment does not make much sense, i probably misunderstood something here aha, and I was just a bit confused by the `lenght(x)`

Reply:
The difference between `length` and `lenghts` is a bit tricky - so, `length(x)` is the number of spectra (returns a single value) while `lengths(x)` is the number of mass peaks per spectrum (returns a vector of length equal to the number of spectra). We want to split and chunk-wise process the `Spectra` object, it makes thus only sense to split if the chunk size is smaller then the length (number of spectra) or a `Spectra`. This is independent of the number of mass peaks per spectrum, we just want to process a smaller subset of our `Spectra` at a time and load only the data for that to reduce the memory demand.